File: 0055_Jump Game/JumpGame.cs
public class Solution {
    public bool CanJump(int[] nums) {
        var furthest = 0;
        for(int i=0;i<nums.Length;i++)
        {
            if(furthest >= i)
            {
                furthest = Math.Max(furthest, nums[i] + i);
                if(furthest >= (nums.Length-1)) break;
            }
            else 
                break;
        }
        
        return furthest >= nums.Length - 1;
    }
}

File: 0055_Jump Game/JumpGame.java
class Solution {
    public boolean canJump(int[] nums) {
        int n = nums.length;
        int furthest = 0;
        
        for(int i=0;i<n;i++) {
            if(i > furthest) return false;
            furthest = Math.max(furthest, nums[i] + i);
            if(furthest >= n-1) return true;
        }
        
        return furthest >= n-1;
    }
}